fixieai/cli/agent/commands.py

```diff
@@ -211,16 +213,35 @@ def _ensure_agent_updated(
     return agent
 
 
-@agent.command("serve", help="Start serving the current agent.")
+@agent.command(
+    "serve", help="Serve the current agent locally via a publicly-accessible URL."
+)
 @click.argument("path", callback=_validate_agent_path, required=False)
 @click.option("--host", default="0.0.0.0")
 @click.option("--port", type=int, default=8181)
+@click.option(
+    "--tunnel/--no-tunnel",
+    is_flag=True,
+    default=True,
+    help="Create a tunnel using localhost.run",
+)
 @click.pass_context
-def serve(ctx, path, host, port):
-    # TODO: add tunneling and deployment url update
+def serve(ctx, path, host, port, tunnel):
+    console = rich_console.Console()
     config, agent_impl = loader.load_agent_from_path(path)
-    agent_api = _ensure_agent_updated(ctx.obj.client, config)
-    agent_impl.serve(agent_api.agent_id, host, port)
+
+    with contextlib.ExitStack() as stack:
+        if tunnel:
+            # Start up a tunnel via localhost.run.
+            console.print(f"Opening tunnel to {host}:{port} via localhost.run.")
+            console.print(
+                f"[red]This will replace any existing deployment until you run [bold]fixie deploy[/bold]![/red]"
```

Comment:
Maybe a bit too much. I'd just say "This replaces any existing deployment, run fixie deploy to redeploy to prod."

Reply:
Sadly, this is where my cutting the deployment API down to the bone is limiting -- once the url is replaced (the deployment becomes "unmanaged") there's no way short of redeploying to tell fixie to manage it again.